## Iterators in python

### Using Iterator and While Loop to mock the behaviour of for loop

In [ ]:
x = list(range(100))
length_of_list_x = len(x)

iter_x = iter(x)
i = 0;
while i <= 10:
    print(next(iter_x), end=',')
    i += 1
    




In [42]:
new_list = list(range(20,220))
print(new_list, end=',')

[20, 21, 22, 23, 24, 25, 26, 27, 28, 29, 30, 31, 32, 33, 34, 35, 36, 37, 38, 39, 40, 41, 42, 43, 44, 45, 46, 47, 48, 49, 50, 51, 52, 53, 54, 55, 56, 57, 58, 59, 60, 61, 62, 63, 64, 65, 66, 67, 68, 69, 70, 71, 72, 73, 74, 75, 76, 77, 78, 79, 80, 81, 82, 83, 84, 85, 86, 87, 88, 89, 90, 91, 92, 93, 94, 95, 96, 97, 98, 99, 100, 101, 102, 103, 104, 105, 106, 107, 108, 109, 110, 111, 112, 113, 114, 115, 116, 117, 118, 119, 120, 121, 122, 123, 124, 125, 126, 127, 128, 129, 130, 131, 132, 133, 134, 135, 136, 137, 138, 139, 140, 141, 142, 143, 144, 145, 146, 147, 148, 149, 150, 151, 152, 153, 154, 155, 156, 157, 158, 159, 160, 161, 162, 163, 164, 165, 166, 167, 168, 169, 170, 171, 172, 173, 174, 175, 176, 177, 178, 179, 180, 181, 182, 183, 184, 185, 186, 187, 188, 189, 190, 191, 192, 193, 194, 195, 196, 197, 198, 199, 200, 201, 202, 203, 204, 205, 206, 207, 208, 209, 210, 211, 212, 213, 214, 215, 216, 217, 218, 219],

## Dynamically print the elements of the list using while loop

1. Initialize an <strong>index variable</strong> which you will pass on the square-bracket syntax(get item syntax)
2. Increament the value of index-variable i until it reaches the end if the index range(199).
3. for each value of index-variable i, pass the index-variable to get-item syntax(square-bracket syntax) to print 4. the individual element of the list

```python
    index = 0

    new_list[index] = 20 when index = 0
Increment index by 1
    new_list[index] = 21 when index = 1


keep incrementing the index until we reach the end, in our case that will be 199
    new_list[index] = 219 when index = 199
```



In [59]:
index = 0


# initialising the value of index to be 0
while index<=20:
# keep on printing the elements of the list until the index is Less Than Equals to 20
    print('index[{}]:{}'.format(index, new_list[index]))
#     print the index and the element on the index
    index += 1
# increament the value of the index variable by 1





print(index)

index[0]:20
index[1]:21
index[2]:22
index[3]:23
index[4]:24
index[5]:25
index[6]:26
index[7]:27
index[8]:28
index[9]:29
index[10]:30
index[11]:31
index[12]:32
index[13]:33
index[14]:34
index[15]:35
index[16]:36
index[17]:37
index[18]:38
index[19]:39
index[20]:40
21


Given a string:
```python
new_string='ajdgfuyasudftasgdfnahdsjgfqrgfhasbfbaleugqergfajdbvnmavckauytefgalsdjvalhjgfyaegfadvnbakjbsdvyqyasdfajsdvcjagsdfuvasdcvbjasuyfgavsdhjfvaveyufajsdfquywegfhqdbfkuqewfgyqvdhcvqkuwegfyuqwevgf'

```

print the characters bettwen index 2 and 120 but only print the elements having even index.

In [62]:
new_string='ajdgfuyasudftasgdfnahdsjgfqrgfhasbfbaleugqergfajdbvnmavckauytefgalsdjvalhjgfyaegfadvnbakjbsdvyqyasdfajsdvcjagsdfuvasdcvbjasuyfgavsdhjfvaveyufajsdfquywegfhqdbfkuqewfgyqvdhcvqkuwegfyuqwevgf'
index = 30
while index <= 100:
    print('index[{}]:{}'.format(index, new_string[index]))
    index += 1

index[30]:h
index[31]:a
index[32]:s
index[33]:b
index[34]:f
index[35]:b
index[36]:a
index[37]:l
index[38]:e
index[39]:u
index[40]:g
index[41]:q
index[42]:e
index[43]:r
index[44]:g
index[45]:f
index[46]:a
index[47]:j
index[48]:d
index[49]:b
index[50]:v
index[51]:n
index[52]:m
index[53]:a
index[54]:v
index[55]:c
index[56]:k
index[57]:a
index[58]:u
index[59]:y
index[60]:t
index[61]:e
index[62]:f
index[63]:g
index[64]:a
index[65]:l
index[66]:s
index[67]:d
index[68]:j
index[69]:v
index[70]:a
index[71]:l
index[72]:h
index[73]:j
index[74]:g
index[75]:f
index[76]:y
index[77]:a
index[78]:e
index[79]:g
index[80]:f
index[81]:a
index[82]:d
index[83]:v
index[84]:n
index[85]:b
index[86]:a
index[87]:k
index[88]:j
index[89]:b
index[90]:s
index[91]:d
index[92]:v
index[93]:y
index[94]:q
index[95]:y
index[96]:a
index[97]:s
index[98]:d
index[99]:f
index[100]:a


In [64]:
new_string='ajdgfuyasudftasgdfnahdsjgfqrgfhasbfbaleugqergfajdbvnmavckauytefgalsdjvalhjgfyaegfadvnbakjbsdvyqyasdfajsdvcjagsdfuvasdcvbjasuyfgavsdhjfvaveyufajsdfquywegfhqdbfkuqewfgyqvdhcvqkuwegfyuqwevgf'
index = 2
while index <= 100:
    print('index[{}]:{}'.format(index, new_string[index]))
    index += 2

index[2]:d
index[4]:f
index[6]:y
index[8]:s
index[10]:d
index[12]:t
index[14]:s
index[16]:d
index[18]:n
index[20]:h
index[22]:s
index[24]:g
index[26]:q
index[28]:g
index[30]:h
index[32]:s
index[34]:f
index[36]:a
index[38]:e
index[40]:g
index[42]:e
index[44]:g
index[46]:a
index[48]:d
index[50]:v
index[52]:m
index[54]:v
index[56]:k
index[58]:u
index[60]:t
index[62]:f
index[64]:a
index[66]:s
index[68]:j
index[70]:a
index[72]:h
index[74]:g
index[76]:y
index[78]:e
index[80]:f
index[82]:d
index[84]:n
index[86]:a
index[88]:j
index[90]:s
index[92]:v
index[94]:q
index[96]:a
index[98]:d
index[100]:a


## excercise

```python
alphabet = 'abcdefghijklmnopqrstuvwxyz'
```
if the character 'k' is present in index 10, then print the index else print the element present in next index.

In [86]:
alphabet = 'abcdefghijalmnopqrstuvwxyz'
# index = 0
# try:
#     while index<=len(alphabet):
#         if alphabet[index] == 'k':
#             print(index)
#         index += 1
        
# except:
#     pass

index=10
if 'k' in alphabet:
    print(index)
else:
    print(alphabet[index+1])



l
